### PR TITLE
Replace git-based MasterSource with CDN-based TrunkSource

### PR DESCRIPTION
**Pull request by igor-makarov**
See [CocoaPods#8923](https://github.com/CocoaPods/CocoaPods/pull/8923) for more info.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 ##### Enhancements
 
+* Replace git-based MasterSource with CDN-based TrunkSource
```
Comment (dnkoutso) [inline CHANGELOG.md]: nit wrap MasterSource and TrunkSource with \`\` for fomatting.

Also need to add two empty spaces at the end of this line for formatting.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,9 @@
...
+  [igor-makarov](https://github.com/igor-makarov)
```
Comment (dnkoutso) [inline CHANGELOG.md]: I know its somewhat counter-intuitive but there is no issue in Core for this, either create an issue and add a link here to it or then just add a link to PR #552.

**Comment by dnkoutso** [comment]
Sorry for the blast of reviewers. I just think one more person should take a look for swapping out the master source! 

This is super exciting.

**Comment by igor-makarov** [comment]
Thank you!

File: CHANGELOG.md (hunk 2 above)
Comment (igor-makarov) [reply CHANGELOG.md]: Shouldn't this just link to CocoaPods#8923?

File: spec/source/manager_spec.rb [hunk 3]
```diff
@@ -25,19 +25,19 @@ module Pod
...
         aggregate = @sources_manager.aggregate_for_dependency(dependency)
-        aggregate.sources.map(&:name).should == %w(master test_cdn_repo_local test_empty_dir_repo test_prefixed_repo test_repo)
+        aggregate.sources.map(&:name).should == %w(artsy test_cdn_repo_local test_empty_dir_repo test_prefixed_repo test_repo test_repo_without_specs_dir trunk)
       end
 
       it 'includes all sources in an aggregate for a dependency if non-existent source is specified' do
         dependency = Dependency.new('JSONKit', '1.4', :source => 'https://url/to/nonexistent/specs.git')
         aggregate = @sources_manager.aggregate_for_dependency(dependency)
-        aggregate.sources.map(&:name).should == %w(master test_cdn_repo_local test_empty_dir_repo test_prefixed_repo test_repo)
+        aggregate.sources.map(&:name).should == %w(artsy test_cdn_repo_local test_empty_dir_repo test_prefixed_repo test_repo test_repo_without_specs_dir trunk)
```
Comment (amorde) [inline spec/source/manager_spec.rb]: is there a reason `trunk` is always last? not that it's an issue, but I think they should be in the order they were declared

File: lib/cocoapods-core/source/manager.rb [hunk 4]
```diff
@@ -394,8 +394,8 @@ def name_for_url(url)
         end
 
         case url.to_s.downcase
-        when %r{github.com[:/]+cocoapods/specs}
-          base = Pod::MasterSource::MASTER_REPO_NAME
+        when Pod::TrunkSource::TRUNK_REPO_URL.downcase
+          base = Pod::TrunkSource::TRUNK_REPO_NAME
```
Comment (amorde) [inline lib/cocoapods-core/source/manager.rb]: could this be updated to handle the trunk URL when it doesn't include the trailing `/`? I think we discussed this in Slack at some point

Comment (dnkoutso) [reply lib/cocoapods-core/source/manager.rb]: +1 on trailing `/` I forgot about that.

File: spec/source/manager_spec.rb (hunk 3 above)
Comment (dnkoutso) [reply spec/source/manager_spec.rb]: It might need to be taken care of yes because lint and Molinillo might discover pods in the order the sources were passed in.

Comment (dnkoutso) [reply spec/source/manager_spec.rb]: but this might not be the place to do this ordering. Maybe it gets re-arranged somewhere else.

Comment (amorde) [reply spec/source/manager_spec.rb]: yeah, but something has changed here because all the test cases now have trunk at the end (where master used to be first). not sure where/why that happened

Comment (igor-makarov) [reply spec/source/manager_spec.rb]: I think this is repo dir sort order.

Comment (dnkoutso) [reply spec/source/manager_spec.rb]: ah yea this looks sorted in the test so I think this is fine.

Comment (igor-makarov) [reply spec/source/manager_spec.rb]: I hate myself for suggesting this, but... Maybe `_trunk`?

Comment (igor-makarov) [reply spec/source/manager_spec.rb]: You know what, never mind.  
If we want to enforce some order for the dep resolver, it has to be done in code related to dep resolution and not in the source manager.

Comment (dnkoutso) [reply spec/source/manager_spec.rb]: correct. I wouldnt bother with the order here.

File: lib/cocoapods-core/source/manager.rb (hunk 4 above)
Comment (igor-makarov) [reply lib/cocoapods-core/source/manager.rb]: Fixed this.

**Comment by igor-makarov** [comment]
Fixed some edge cases in CDN URL handling + added a few tests for the trailing slashes/newlines in CDN URL.

**Comment by dnkoutso** [comment]
Please rebase now that 1.7.3 is released and ensure `CHANGELOG` entry is at the right section (master)

**Comment by igor-makarov** [comment]
Rebased Core, CocoaPods and Integration. 💪

**Comment by dnkoutso** [comment]
@igor-makarov just need a rebase and we can land.

**Comment by igor-makarov** [comment]
Done!
